pytests/test_bjoern.py

```diff
@@ -0,0 +1,464 @@
+        os.kill(os.getpid(), signal.SIGINT)
+        yield b'chunk2'
+        yield b'chunk3'
+
+    client.start(application)
+    assert client.get().content == b'chunk1chunk2chunk3'
+
+
+def test_exc_info_reference(client):
+    """tests/test_exc_info_reference.py"""
+
+    def app(env, start_response):
+        start_response('200 alright', [])
+        try:
+            a
+        except:
+            import sys
+
+            x = sys.exc_info()
+            start_response('500 error', [], x)
+        return [b'hello']
+
+    client.start(app)
+    response = client.get()
+    assert response.status_code == 500
+    assert response.content == b'hello'
+
+
+def test_fork(unixclient):
+    """tests/fork.py"""
+
+    def app(env, start_response):
+        start_response('200 ok', [])
+        return str(os.getpid()).encode('ascii')
+
+    unixclient.start(app, num_workers=10)
+    pids = {unixclient.get().text for _ in range(100)}
+    assert len(pids) > 1
+
+
+def test_wsgi_compliance(client):
+    """test_wsgi_compliance.py"""
+
+    @validator
+    def _app(environ, start_response):
+        start_response('200 OK', [('Content-Type', 'text/plain')])
+        return [b'Hello World']
+
+    client.start(_app)
+    client.get()
+
+
+@pytest.mark.parametrize(
+    'expect_header_value,content_length,body,response',
+    (
+        ('100-continue', 300, '', b'HTTP/1.1 100 Continue\r\n\r\n'),
+        (
+            '100-continue',
+            0,
+            '',
+            b'HTTP/1.1 200 OK\r\nContent-Length: 0\r\nConnection: Keep-Alive\r\n\r\n',
+        ),
+        (
+            '100-continue',
+            4,
+            'test',
+            b'HTTP/1.1 200 OK\r\nContent-Length: 0\r\nConnection: Keep-Alive\r\n\r\n',
+        ),
+        ('badness', 0, '', b'HTTP/1.1 417 Expectation Failed\r\n\r\n'),
+        ('badness', 300, '', b'HTTP/1.1 417 Expectation Failed\r\n\r\n'),
+        ('badness', 4, 'test', b'HTTP/1.1 417 Expectation Failed\r\n\r\n'),
+    ),
+)
+def test_expect_100_continue(
+    httpclient, expect_header_value, content_length, body, response
+):
+    """tests/expect100.py"""
+
+    def app(e, s):
+        s('200 OK', [('Content-Length', '0')])
+        return b''
+
+    try:
+        socket.SO_REUSEPORT
+    except AttributeError:
+        httpclient.reuse_port = False
+    else:
+        httpclient.reuse_port = True
+
+    httpclient.start(app)
+    # requests doesn't support expect100, see https://github.com/psf/requests/issues/3614
+    # use the raw connection instead of httpclient
+    with contextlib.closing(
+        socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    ) as raw_client:
+        raw_client.connect((httpclient.host, httpclient.port))
+        request = 'GET /fizz HTTP/1.1\r\nExpect: {}\r\nContent-Length: {}\r\n\r\n{}'.format(
+            expect_header_value, content_length, body
+        )
+        raw_client.send(request.encode('utf-8'))
+        resp = raw_client.recv(1 << 10)
```

Comment:
Why not simply write 1024 here

Reply:
This is due to code copying from `tests/expect100.py`. Shouldn't it even be `bjoern.DEFAULT_LISTEN_BACKLOG`?

Reply:
The reason I did that in expect100.py (I'm the author of that specific test) is because I consider it more legible when dealing with byte counts between KB -> MB -> GB. E.g.,

KB = 1024
MB = 1048576
GB = 1073741824

vs.

KB = 1 << 10
MB = 1 << 20
GB = 1 << 30

or, if your test grows the content-length exponentially with each iteration, such as that test does [here](https://github.com/jonashaag/bjoern/blob/master/tests/expect100.py#L45)